### PR TITLE
Do not assume --yes was present when using pulumi in non-interactive mode

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/pulumi/pulumi/issues/3435

This is going into the 2.0 branch.

### CODE REVIEW

**Comment by lukehoban** [comment]
Can we add to https://github.com/pulumi/pulumi/issues/3751?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,15 @@ CHANGELOG
 =========
 
 ## HEAD (Unreleased)
+
+- CLI behavior change.  Commands in non-interactive mode (i.e. when `pulumi` has its output piped to
+  another process) will not default to assuming that `--yes` was passed in.  `--yes` is not
+  explicitly required to proceed in non-interactive scenarios. This affects
```
Comment (lukehoban) [inline CHANGELOG.md]: Is now?

Comment (CyrusNajmabadi) [reply CHANGELOG.md]: Fixed.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,15 @@ CHANGELOG
 =========
 
 ## HEAD (Unreleased)
+
+- CLI behavior change.  Commands in non-interactive mode (i.e. when `pulumi` has its output piped to
+  another process) will not default to assuming that `--yes` was passed in.  `--yes` is now
```
Comment (SimenB) [inline CHANGELOG.md]:
```suggestion
  another process or running on CI) will not default to assuming that `--yes` was passed in.  `--yes` is now
```

something like that, so people running pulumi on CI/in scripts will think "oh, this affects us"?

Comment (CyrusNajmabadi) [reply CHANGELOG.md]: Sounds great.  have merged that in. thanks!

**Pull request merged**